examples/2d/texture_atlas.rs

```diff
@@ -8,11 +8,9 @@ fn main() {
         .init_resource::<RpgSpriteHandles>()
         .add_plugins(DefaultPlugins.set(ImagePlugin::default_nearest())) // prevents blurry sprites
         .add_state::<AppState>()
-        .add_systems((
-            load_textures.in_schedule(OnEnter(AppState::Setup)),
-            check_textures.in_set(OnUpdate(AppState::Setup)),
-            setup.in_schedule(OnEnter(AppState::Finished)),
-        ))
+        .add_systems(OnEnter(AppState::Setup), load_textures)
+        .add_systems(Update, check_textures.in_set(OnUpdate(AppState::Setup)))
```

Comment:
> Definitely makes me want to remove the OnUpdate sets in favor of run_if(in_state(X)).

oh, and that also would be easier to understand IMO, since there would be no misconception of "why is the same Update word used twice here in the same line, how are they related?" (when they are *not* related)

Reply:
Yup if we can implement run_if in a satisfying way for tuples of systems, I'm on board (see the follow ups in the PR description).